worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/factory/parser/transform/DefaultTransformParser.java

```diff
@@ -0,0 +1,136 @@
+import com.fastasyncworldedit.core.extent.transform.MultiTransform;
+import com.fastasyncworldedit.core.extent.transform.RandomTransform;
+import com.fastasyncworldedit.core.math.random.TrueRandom;
+import com.fastasyncworldedit.core.util.StringMan;
+import com.sk89q.minecraft.util.commands.CommandLocals;
+import com.sk89q.worldedit.WorldEdit;
+import com.sk89q.worldedit.extension.input.InputParseException;
+import com.sk89q.worldedit.extension.input.NoMatchException;
+import com.sk89q.worldedit.extension.input.ParserContext;
+import com.sk89q.worldedit.extension.platform.Actor;
+import com.sk89q.worldedit.internal.expression.Expression;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+public class DefaultTransformParser extends FaweParser<ResettableExtent> {
+
+    public DefaultTransformParser(WorldEdit worldEdit) {
+        super(worldEdit);
+    }
+
+    @Override
+    public ResettableExtent parseFromInput(String input, ParserContext context) throws InputParseException {
+        if (input.isEmpty()) {
+            return null;
+        }
+
+        List<Double> unionChances = new ArrayList<>();
+        List<Double> intersectionChances = new ArrayList<>();
+
+        List<ResettableExtent> intersection = new ArrayList<>();
+        List<ResettableExtent> union = new ArrayList<>();
+        final CommandLocals locals = new CommandLocals();
+        Actor actor = context != null ? context.getActor() : null;
+        if (actor != null) {
+            locals.put(Actor.class, actor);
+        }
+        try {
+            List<Map.Entry<ParseEntry, List<String>>> parsed = parse(input);
+            for (Map.Entry<ParseEntry, List<String>> entry : parsed) {
+                ParseEntry pe = entry.getKey();
+                String command = pe.getInput();
+                ResettableExtent transform = null;
+                double chance = 1;
+                if (command.isEmpty()) {
+                    transform = parseFromInput(StringMan.join(entry.getValue(), ','), context);
+                } else if (!worldEdit.getTransformFactory().containsAlias(command)) {
+                    // Legacy syntax
+                    int percentIndex = command.indexOf('%');
+                    if (percentIndex != -1) {  // Legacy percent pattern
+                        chance = Expression.compile(command.substring(0, percentIndex)).evaluate();
+                        command = command.substring(percentIndex + 1);
+                        if (!entry.getValue().isEmpty()) {
+                            if (!command.isEmpty()) {
+                                command += " ";
+                            }
+                            command += StringMan.join(entry.getValue(), " ");
+                        }
+                        transform = parseFromInput(command, context);
+                    } else {
+                        throw new NoMatchException("See: //transforms");
```

Comment:
Deprecated in favor of components. That is not a command either. See my comment on `//masks`.